src/deps.ts

```diff
@@ -296,13 +296,36 @@ export interface AutoEncryptionOptions {
     /** Command line arguments to use when auto-spawning a mongocryptd */
     mongocryptdSpawnArgs?: string[];
     /**
-     * Full path to a CSFLE shared library to be used (instead of mongocryptd)
-     * @experimental
+     * Full path to a CSFLE shared library to be used (instead of mongocryptd).
+     *
+     * This needs to be the path to the file itself, not a directory.
+     * It can be an absolute or relative path. If the path is relative and
+     * its first component is `$ORIGIN`, it will be replaced by the directory
+     * containing the mongodb-client-encryption native addon file. Otherwise,
+     * the path will be interpreted relative to the current working directory.
+     *
+     * Currently, loading different CSFLE shared library files from different
+     * MongoClients in the same process is not supported.
+     *
+     * If this option is provided and no CSFLE shared library could be loaded
+     * from the specified location, creating the MongoClient will fail.
+     *
+     * If this option is not provided and `csfleRequired` is not specified,
+     * the AutoEncrypter will attempt to spawn and/or use mongocryptd according
+     * to the mongocryptd-specific `extraOptions` options.
      */
     csflePath?: string;
+    /**
+     * If specified, never use mongocryptd and instead fail when the CSFLE shared library
+     * could not be loaded.
+     *
+     * This is implied when `csflePath` is specified.
```

Comment:
```suggestion
     * This is always true when `csflePath` is specified. Specifying a path prevents mongocryptd from being used as a fallback.
```
Does this mean I cannot provide a path and still expect mongocryptd to work as a fallback? That makes sense to me since the users of the lib are folks who want to move off of or can't use cryptd, but I'm thinking we could make it clear. 
My suggested text is  maybe in the wrong spot, maybe it would be good to add after `loading different CSFLE shared library files from different MongoClients in the same process is not supported.` Thoughts?

Reply:
Sure – I’ve split the sentence and added the second part to `csflePath` rather than this option, I think that makes sense :+1: